desc/builder/enum.go

```diff
@@ -250,20 +250,17 @@ func (eb *EnumBuilder) buildProto(path []int32, sourceInfo *descriptorpb.SourceC
 
 	if len(needNumbersAssigned) > 0 {
 		tags := make([]int, len(values)-len(needNumbersAssigned))
-		for i, ev := range values {
+		tagsIndex := 0
+		for _, ev := range values {
 			tag := ev.GetNumber()
 			if tag != 0 {
-				tags[i] = int(tag)
+				tags[tagsIndex] = int(tag)
+				tagsIndex++
 			}
 		}
```

Comment:
Oops, thanks for fixing the use of wrong index. However, I think there is still a bug in here. This logic uses `tag != 0` as the guard for whether a tag is in use. But it really should be looping over the builders (i.e. `range eb.values`) and then looking at `evb.numberSet`, like in the above loop.

Another option is to combine this loop with the one above. But I think I carved it out separate so that if you always specify the numbers for every value, then there's a happy path with no extra allocations.